scss/_patterns_equal-height-row.scss

```diff
@@ -0,0 +1,194 @@
+    &.has-3rd-divider:not(.has-1st-divider) .p-equal-height-row__col::before,
+    &.has-3rd-divider:not(.has-2nd-divider) .p-equal-height-row__col::after {
+      @content;
+    }
+  }
+}
+
+@mixin vf-p-equal-height-row {
+  .p-equal-height-row {
+    @extend %vf-row;
+    position: relative;
+
+    .p-equal-height-row__col {
+      // smaller screens each column will have border top by default
+      border-top-color: $colors--theme--border-default;
+      border-top-style: solid;
+      border-top-width: 1px;
+      display: grid;
+      grid-column: span $grid-columns-small;
+      grid-row: span 4;
+      grid-template-rows: subgrid;
+      margin-top: -1px;
+      position: relative;
+
+      @media screen and ($breakpoint-small <= width < $breakpoint-large) {
+        grid-column: span $grid-columns-medium;
+        grid-template-columns: subgrid;
+
+        // for medium screen, each column item will take half of the available cols from the parent grid
+        .p-equal-height-row__item {
+          grid-column: span calc($grid-columns-medium / 2);
+        }
+
+        // for medium screen, position the first column item on the left of the grid
+        .p-equal-height-row__item:first-child {
+          grid-row: span 3;
+        }
+      }
+
+      @media screen and (width >= $breakpoint-large) {
+        border: none;
+        grid-column: span calc($grid-columns / 4);
+      }
+    }
+
+    // divider styles
+    @include divider-styles($large-screen: false) {
+      @extend %vf-pseudo-border;
+    }
+
+    // remove row level dividers for smaller screen sizes
+    @include divider-styles($large-screen: true) {
+      @extend %vf-pseudo-border;
+      display: none;
+    }
+
+    @include position-divider-by-order-in-grid(1) {
+      grid-row: 2;
+    }
+
+    @include position-divider-by-order-in-grid(2) {
+      grid-row: 3;
+    }
+
+    @include position-divider-by-order-in-grid(3) {
+      grid-row: 4;
+    }
+
+    @media screen and ($breakpoint-small <= width < $breakpoint-large) {
+      // We don't need to insert divider below item-1 for medium screen size since item-1 gets positioned on the left
+      @include position-divider-by-order-in-grid(1) {
+        display: none;
+      }
+
+      @include position-divider-by-order-in-grid(2) {
+        grid-column: 4 / 7;
+        grid-row: 2;
+      }
+
+      @include position-divider-by-order-in-grid(3) {
+        grid-column: 4 / 7;
+        grid-row: 3;
+      }
+    }
+
+    @media screen and (width >= $breakpoint-large) {
+      // remove column level dividers for large screen sizes
+      @include divider-styles($large-screen: false) {
+        display: none;
+      }
+
+      @include divider-styles($large-screen: true) {
+        display: block;
+        margin: auto;
+        // for a full 12 col grid, there are default padding that must be taken into consideration by the divider
+        width: calc(100% - 2 * map-get($grid-margin-widths, default));
```

Comment:
Would something like that work instead? To prevent calculating the padding in, and just rely on the grid?

```
grid-column-start: 1;
grid-column-end: span 12;
```

Reply:
I think this would be the same as `grid-column: span 12`, which is what I tested. The issue with this is that the divider will span the correct number of parent grid columns, but the pseudo element is absolutely positioned, so it does not respect the parent grid padding.

Reply:
As far as I can see `grid-column: span 12` relies on auto positioning. 
If you specify `grid-column-start: 1` it seems to set the start position correctly at the start of first column, and the width can be achieved with `grid-column-end: span 12`. 

<img width="1910" alt="image" src="https://github.com/canonical/vanilla-framework/assets/83575/88bee54c-621c-4291-b4cb-b1cf9b96c908">

Although this may be relying on some grid CSS edge case usage with position absolute

Reply:
Aaah I see, this definitely works, so it seems that if we explicitly define `grid-column-start` and `grid-column-end` then the divider will align against the grid column tracks. Setting `grid-column-end` to `13` also works, but I think `span 12` seems to be a bit clearer